upload/helpers.py

```diff
@@ -637,16 +638,39 @@ def dispatch_upload_task(task_arguments, repository, redis):
         timezone.now().timestamp(),
     )
 
-    # Send task to worker
-    TaskService().upload(
+    commitid = task_arguments.get("commit")
+
+    upload_sig = TaskService().upload_signature(
         repoid=repository.repoid,
-        commitid=task_arguments.get("commit"),
+        commitid=commitid,
         report_code=task_arguments.get("report_code"),
-        countdown=max(
-            countdown, int(get_config("setup", "upload_processing_delay") or 0)
-        ),
+        # this prevents the results of the notify task (below) from being
+        # passed as args to this upload task
+        immutable=True,
     )
 
+    notified = CommitNotification.objects.filter(commit__commitid=commitid).exists()
```

Comment:
I'm realizing we probably need to store this per-PR since when we make another commit on the same branch we don't necessarily want to show the "processing" comment again. 🤔